connector/roundrobinconnector/connector.go

```diff
@@ -0,0 +1,86 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package roundrobinconnector // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/roundrobinconnector"
+
+import (
+	"context"
+	"sync/atomic"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+func allConsumers[T any](r router[T]) ([]T, error) {
+	pipeIDs := r.PipelineIDs()
+	consumers := make([]T, len(pipeIDs))
+	for i, pipeID := range pipeIDs {
+		cons, err := r.Consumer(pipeID)
+		if err != nil {
+			return nil, err
+		}
+		consumers[i] = cons
+	}
+	return consumers, nil
+}
+
+type router[T any] interface {
+	PipelineIDs() []component.ID
+	Consumer(pipelineIDs ...component.ID) (T, error)
+}
```

Comment:
Should this be how we implement it in core?

Reply:
You mean to help remove duplicate code? I can craft a quick PR.